hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/validators/TokenUpdateValidator.java

```diff
@@ -62,6 +63,10 @@ public ValidationResult validateSemantics(
         if (op.hasMemo()) {
             context.attributeValidator().validateMemo(op.memo());
         }
+        // validate metadata
+        if (op.metadata() != null && !Arrays.isNullOrEmpty(op.metadata().toByteArray())) {
```

Comment:
I think we should just check null. If metadata is present and is empty we should set empty in my opinion